### PR TITLE
[80_6] Improving Radical Box Typesetting

### PR DESCRIPTION
**Pull request by KeShih**
## What
Improving Fraction Box Typesetting.

## Why
For some opentype math font, math table gives more precise information about typesetting radical.
 
## How to test your changes?
Open `devel/80_6.tmu`.

It is evident that the Asana Math and Fira Math opentype fonts have undergone changes, with significant adjustments in the spacing and positioning before and after the degree symbol. In contrast, the latter two hardcoded opentype fonts have remained unchanged.

Before:
image

After:
image

### CODE REVIEW

**Review by da-liii — APPROVED** [review]
LGTM

File: src/Typeset/Boxes/Composite/math_boxes.cpp [hunk 1]
```diff
@@ -191,6 +193,12 @@ sqrt_box_rep::sqrt_box_rep (path ip, box b1, box b2, box sqrtb, font fn2,
       else if (occurs ("agella", fn->res_name)) Y+= (16 * bw) >> 3;
       else Y+= (15 * bw) >> 3;
     }
+    else if (use_open_type) {
+      Y+= fn->sqrt_degree_rise_percent * sqrtb->h () / 100;
+      M  = fn->sqrt_kern_after_degree;
+      sep= 0;
+      b2->x1-= fn->sqrt_kern_before_degree;
```
Comment (da-liii) [inline src/Typeset/Boxes/Composite/math_boxes.cpp:200]: a bit confused on this parameter

Comment (da-liii) [reply src/Typeset/Boxes/Composite/math_boxes.cpp:200]: image

**Comment by da-liii** [comment]
image

The rule of the radical box rendered in Fira Math font should be fixed

File: src/Typeset/Boxes/Composite/math_boxes.cpp (hunk 1 above)
Comment (KeShih) [reply src/Typeset/Boxes/Composite/math_boxes.cpp:200]: The OpenType math table requires spacing both before and after the degree symbol. In the case of b2->x1, the front spacing is added, while M represents the distance from the back of the symbol to the radical. According to the specification, the height of the degree symbol is raised by a fixed proportion of the height of the radical.

File: src/Graphics/Fonts/font.cpp [hunk 2]
```diff
@@ -109,6 +109,13 @@ font_rep::copy_math_pars (font fn) {
   frac_denom_disp_shift_down   = fn->frac_denom_disp_shift_down;
   frac_denom_gap_min           = fn->frac_denom_gap_min;
   frac_denom_disp_gap_min      = fn->frac_denom_disp_gap_min;
+  sqrt_ver_gap                 = fn->sqrt_ver_gap;
+  sqrt_ver_disp_gap            = fn->sqrt_ver_disp_gap;
+  sqrt_rule_thickness          = fn->sqrt_rule_thickness;
```
Comment (da-liii) [inline src/Graphics/Fonts/font.cpp:114]: unused

**Pull request merged**